groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -4499,39 +4547,48 @@ bool operator<(const bsl::optional<LHS_TYPE>& lhs,
 template <class LHS_TYPE, class RHS_TYPE>
 bool operator<(const std::optional<LHS_TYPE>& lhs,
                const bsl::optional<RHS_TYPE>& rhs);
-    // If !y, false; otherwise, if !x, true; otherwise *x < *y.  Note that this
-    // function will fail to compile if 'LHS_TYPE' and 'RHS_TYPE' are not
-    // compatible.
+    // Return 'true' if the specified 'lhs' optional object is ordered before
+    // the specified 'rhs' optional object, and 'false' otherwise.  'lhs' is
+    // ordered before 'rhs' if 'lhs' is disengaged and 'rhs' is engaged or if
+    // both are engaged and 'lhs.value()' is ordered before 'rhs.value()'.
+    // Note that this function will fail to compile if 'LHS_TYPE' and
+    // 'RHS_TYPE' are not compatible.
 
 template <class LHS_TYPE, class RHS_TYPE>
 bool operator>(const bsl::optional<LHS_TYPE>& lhs,
                const std::optional<RHS_TYPE>& rhs);
 template <class LHS_TYPE, class RHS_TYPE>
 bool operator>(const std::optional<LHS_TYPE>& lhs,
                const bsl::optional<RHS_TYPE>& rhs);
-    // If !x, false; otherwise, if !y, true; otherwise *x > *y. note that this
-    // function will fail to compile if 'LHS_TYPE' and 'RHS_TYPE' are not
-    // compatible.
+    // Return 'true' if the specified 'lhs' optional object is ordered after
+    // the specified 'rhs' optional object, and 'false' otherwise.  'lhs' is
+    // ordered after 'rhs' if 'lhs' is engaged and 'rhs' is disengaged or if
+    // both are engaged and 'lhs.value()' is ordered after 'rhs.value()'.  Note
+    // that this operator returns 'rhs < lhs'.  Also note that this function
+    // will fail to compile if 'LHS_TYPE' and 'RHS_TYPE' are not compatible.
 
 template <class LHS_TYPE, class RHS_TYPE>
 bool operator<=(const bsl::optional<LHS_TYPE>& lhs,
                 const std::optional<RHS_TYPE>& rhs);
 template <class LHS_TYPE, class RHS_TYPE>
 bool operator<=(const std::optional<LHS_TYPE>& lhs,
                 const bsl::optional<RHS_TYPE>& rhs);
-    // If !x, true; otherwise, if !y, false; otherwise *x <= *y. Note that this
-    // function will fail to compile if 'LHS_TYPE' and 'RHS_TYPE' are not
-    // compatible.
+    // Return 'true' if the specified 'lhs' is ordered before the specified
+    // 'rhs' optional object or 'lhs' and 'rhs' have the same value, and
+    // 'false' otherwise.  (See 'operator<' and 'operator=='.)  Note that this
+    // operator returns '!(rhs < lhs)'.
```

Comment:
`Also note that this function will fail to compile if 'LHS_TYPE' and 'RHS_TYPE' are not compatible.` was lost. Intentional?

Reply:
not intentional. Fixed.